tornado/testing.py

```diff
@@ -201,41 +190,12 @@ def setUp(self) -> None:
             module=r"tornado\..*",
         )
         super().setUp()
-        # NOTE: this code attempts to navigate deprecation warnings introduced
-        # in Python 3.10. The idea of an implicit current event loop is
-        # deprecated in that version, with the intention that tests like this
-        # explicitly create a new event loop and run on it. However, other
-        # packages such as pytest-asyncio (as of version 0.16.0) still rely on
-        # the implicit current event loop and we want to be compatible with them
-        # (even when run on 3.10, but not, of course, on the future version of
-        # python that removes the get/set_event_loop methods completely).
-        #
-        # Deprecation warnings were introduced inconsistently:
-        # asyncio.get_event_loop warns, but
-        # asyncio.get_event_loop_policy().get_event_loop does not. Similarly,
-        # none of the set_event_loop methods warn, although comments on
-        # https://bugs.python.org/issue39529 indicate that they are also
-        # intended for future removal.
-        #
-        # Therefore, we first attempt to access the event loop with the
-        # (non-warning) policy method, and if it fails, fall back to creating a
-        # new event loop. We do not have effective test coverage of the
-        # new event loop case; this will have to be watched when/if
-        # get_event_loop is actually removed.
-        self.should_close_asyncio_loop = False
         try:
-            self.asyncio_loop = asyncio.get_event_loop_policy().get_event_loop()
-        except Exception:
-            self.asyncio_loop = asyncio.new_event_loop()
-            self.should_close_asyncio_loop = True
-
-        async def get_loop() -> IOLoop:
-            return self.get_new_ioloop()
-
-        self.io_loop = self.asyncio_loop.run_until_complete(get_loop())
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore", DeprecationWarning)
-            self.io_loop.make_current()
+            self._prev_aio_loop = asyncio.get_event_loop()
```

Comment:
I think this maintains compatibility with what pytest-asyncio does? At least, it passes the test that appears to be checking this.

It will create an unnecessary event loop when there isn't already a current one. But after the first test has done this, there will be a current event loop, so subsequent tests will only create and tear down the one they're using.

Reply:
This is sneaky - it works in tornado's test suite because we configure deprecation warnings to be errors. But in a default configuration, it will log a warning and there's no good way to suppress it. (This makes me think that we probably need a CI configuration that doesn't have warnings-as-errors, just like we have one that runs in -O mode to ensure we don't rely on the assert statement which can be disabled). 

If the pytest-asyncio integration were valuable enough I think we could figure out a way to keep it and deal with the warning, but given the precedent of IsolatedAsyncioTestCase I think we're better off just simplifying it. The person who originally asked for this feature in #2324 says they no longer need it, and I suspect that the reason they needed it in the first place was something idiosyncratic about their application rather than something broadly applicable to all pytest users.

Reply:
In this `setUp` method, we're specifically silencing those deprecation warnings (if triggered by tornado code). As I understand the filtering system, this should hide them for anyone using AsyncTestCase, regardless of what options they set in their own code.

However, if it's no longer needed now, we can simplify the code to just clear the event loop, without storing the previous one. I think that means taking out `GetNewIOLoopTest` as well.